bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/CloudBigtableConnectionPool.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.dataflow;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.Connection;
+
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import com.google.cloud.bigtable.hbase1_0.BigtableConnection;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.LinkedHashMultimap;
+
+/**
+ * Pubsub and other windowed sources can have a large quantity of bundles in short amounts of time.
+ * {@link CloudBigtableIO.AbstractCloudBigtableTableWriteFn} should not create a connection per
+ * bundle, since that could happen ever few milliseconds. Rather, the should rely on a connection
+ * pool to better manage connection lifespans.
+ */
+public class CloudBigtableConnectionPool {
```

Comment:
I moved the static instance to CloudBigtableIO.AbstractCloudBigtableTableWriteFn where it seems to belong.